design/BufferOperations.md

```diff
@@ -156,7 +159,7 @@ function bufferSubData(device, destBuffer, destOffset, srcArrayBuffer) {
     const byteCount = srcArrayBuffer.byteLength;
     const (srcBuffer, mapping) = device.createBufferMapped({
         size: byteCount,
-        usage: GPUBufferUsage.MAP_WRITE | GPUBufferUsage.TRANSFER_SRC,
```

Comment:
This shim would still keep `MAP_WRITE` to avoid having both a staging and GPU local allocation for the staging buffer.

Reply:
Sorry, I don't totally understand. It doesn't seem like this shim uses the MAP_WRITE usage at all, so why does it help? (Implementation can choose whatever internal usages it wants.)

Reply:
Oh nvm :)